components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpProducer.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * 
+ */
+package org.apache.camel.component.snmp;
+
+
+import org.apache.camel.Exchange;
+import org.apache.camel.impl.DefaultProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snmp4j.CommunityTarget;
+import org.snmp4j.PDU;
+import org.snmp4j.Snmp;
+import org.snmp4j.TransportMapping;
+import org.snmp4j.event.ResponseEvent;
+import org.snmp4j.mp.MPv3;
+import org.snmp4j.security.SecurityModels;
+import org.snmp4j.security.SecurityProtocols;
+import org.snmp4j.security.USM;
+import org.snmp4j.smi.Address;
+import org.snmp4j.smi.GenericAddress;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.VariableBinding;
+import org.snmp4j.transport.DefaultTcpTransportMapping;
+import org.snmp4j.transport.DefaultUdpTransportMapping;
+
+/**
+ * A snmp producer
+ * 
+ * @author FabryProg <fabryprog@gmail.com>
+ *
+ */
+public class SnmpProducer extends DefaultProducer {
+   
+    private static final Logger LOG = LoggerFactory.getLogger(SnmpProducer.class);
+    
+    private SnmpEndpoint endpoint;
+    private TransportMapping<? extends Address> transport;
+    private Snmp snmp;
+    private USM usm;
+    private CommunityTarget target;
+    private PDU pdu;
+    private Address targetAddress;
+
```

Comment:
Most of these instance variables are initialized, used, released inside the method ```process``` so the scope can be reduced to method scope otherwise you may have threading issues.